### PR TITLE
File-dropzone: Add support for disabled state

### LINKED ISSUE #847: File Dropzone: Add support for `disabled` state
Currently, there is no property to set a disabled state for the component.

### PR DESCRIPTION
**Pull request by prashanthr6383**
## Description

References 
Fixes #847 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

https://deploy-preview-2902--moduswebcomponents.netlify.app/?path=/story/components-file-dropzone--default&args=disabled:true

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.tsx [hunk 1]
```diff
@@ -52,6 +52,9 @@ export class ModusFileDropzone {
   /** (optional) Whether multiple files can be uploaded. */
   @Prop() multiple = true;
 
+  /** (optional) disables the dropzone*/
+  @Prop({ reflect: true }) disabled: boolean;
```
Comment (prashanth-offcl) [inline stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.tsx:56]: @prashanthr6383 Can you make sure the drop behavior is also handled for the disabled state.

**Comment by coliff** [comment]
Please ensure dark mode styles have been fixed:
image

**Comment by prashanthr6383** [comment]
> Please ensure dark mode styles have been fixed: image

Hi @coliff, I have added the dark mode styles. please check
image

File: stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.scss [hunk 2]
```diff
@@ -34,13 +34,12 @@
     color: $modus-file-upload-dropzone-color;
     display: flex;
     flex-direction: column;
-    font-weight: $font-weight-semi-bold;
+    font-weight: 400;
```
Comment (prashanth-offcl) [inline stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.scss:37]: we can remove the font weight as 400 / normal is the default value

**Comment by prashanthr6383** [comment]
@prashanth-offcl accidently removed the comment. I have removed the font-weight property as you mentioned

**Review by prashanth-offcl — APPROVED** [review]

**Pull request merged**